src/lib/Microsoft.Fx.Portability/ApiPortService.cs

```diff
@@ -47,126 +51,203 @@ public ApiPortService(string endpoint, HttpMessageHandler httpMessageHandler, Pr
-                var defaultFormat = await GetDefaultResultFormatAsync();
-                return new[] { defaultFormat.Response };
+                case HttpStatusCode.BadRequest:
+                    await ProcessBadRequestAsync(response);
+                    break; // ProcessBadRequestAsync always throws but the compiler does not detect it
+                case HttpStatusCode.MovedPermanently:
+                    throw new MovedPermanentlyException();
+                case HttpStatusCode.NotFound:
+                    // Estimated maximum allowed content from the portability service in bytes
+                    const long estimatedMaximumAllowedContentLength = 31457280;
+
+                    var contentLength = request.Content?.Headers?.ContentLength;
+
+                    if (contentLength.HasValue && contentLength.Value >= estimatedMaximumAllowedContentLength)
+                    {
+                        throw new RequestTooLargeException(contentLength.Value);
+                    }
+                    else
+                    {
+                        throw new NotFoundException(request.Method, request.RequestUri);
+                    }
+                case HttpStatusCode.Unauthorized:
+                    throw new UnauthorizedEndpointException();
+                case HttpStatusCode.ProxyAuthenticationRequired:
+                    throw new ProxyAuthenticationRequiredException(request.RequestUri);
             }
-            else
-            {
-                var requestedFormats = new HashSet<string>(formats, StringComparer.OrdinalIgnoreCase);
-                var resultFormats = await GetResultFormatsAsync();
-                var formatInformation = resultFormats.Response
-                    .Where(r => requestedFormats.Contains(r.DisplayName));
 
-                var unknownFormats = requestedFormats
-                    .Except(formatInformation.Select(f => f.DisplayName), StringComparer.OrdinalIgnoreCase);
+            throw new PortabilityAnalyzerException(string.Format(CultureInfo.CurrentCulture, LocalizedStrings.UnknownErrorCodeMessage, response.StatusCode));
+        }
 
-                if (unknownFormats.Any())
-                {
-                    throw new UnknownReportFormatException(unknownFormats);
-                }
+        private static async Task ProcessBadRequestAsync(HttpResponseMessage response)
+        {
+            var content = await response.Content.ReadAsStringAsync();
 
-                return formatInformation;
+            if (string.Equals(response.ReasonPhrase, typeof(UnknownTargetException).Name, StringComparison.Ordinal))
+            {
+                throw new UnknownTargetException(content);
             }
+
+            throw new PortabilityAnalyzerException(LocalizedStrings.BadRequestMessage);
+        }
+
+        public Task<IReadOnlyCollection<ApiDefinition>> SearchFxApiAsync(string query, int? top = null)
+        {
+            throw new NotImplementedException();
```

Comment:
This one, I think we do need. We may be able to drop the other FxApi-related service methods, but I believe this is used from `ApiPort.exe docId` which we should continue supporting.

Reply:
Yes, this one's reachable, but I'm inclined not to implement it for the new service because I think it's superseded by apisof.net.